scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -115,6 +114,11 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
 }
 
 object BeamWriter {
+  // We create BeamWriter per Execution. To share forcedPipes across executions we are putting
```

Comment:
Can it have some kind of unexpected behavior?

Reply:
yes. Why did you move it out of `BeamWriter`? The idea is all mutable state lives in the `Writer` for the scope of an execution.

We do not want to have these survive past a Writer lifetime.

Reply:
When I tested it out I was not able to make it work.
In the run method of `Execution` we create a new writer. Now lets say I do
`val tp1 = TypedPipe(..).forceToDiskExecution`
`val tp2 = tp1.flatmap(..)` 

How do we share forcedPipe info across executions?

Reply:
note:

https://github.com/twitter/scalding/blob/1312b22b389b6ae226be6e7ab9a3af44bac2fde5/scalding-core/src/main/scala/com/twitter/scalding/Execution.scala#L151

we create a single `Writer` for the life of the final `Execution` evaluation. So, that `run` method is only called by `waitFor` in your code.

We do not want to share state across two `waitFor` calls. We actually should have tests that we *don't* do that.

If you want to share, you put it inside one `Execution` that's the whole point of `Execution`: creating an environment to model sequential and shared state. But after it is done, the only effects you should see are the things you finally wrote out and the result value of the `Execution`.